server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -277,4 +278,44 @@ public void testRemoteSegmentCleanup() throws Exception {
     public void testRemoteTranslogCleanup() throws Exception {
         verifyRemoteStoreCleanup(true);
     }
+
+    public void testStaleCommitDeletionWithInvokeFlush() throws Exception {
+        internalCluster().startDataOnlyNodes(3);
+        createIndex(INDEX_NAME, remoteStoreIndexSettings(1, 10000l));
+        int numberOfIterations = randomIntBetween(5, 15);
+        indexData(numberOfIterations, true);
+        String indexUUID = client().admin()
+            .indices()
+            .prepareGetSettings(INDEX_NAME)
+            .get()
+            .getSetting(INDEX_NAME, IndexMetadata.SETTING_INDEX_UUID);
+        Path indexPath = Path.of(String.valueOf(absolutePath), indexUUID, "/0/segments/metadata");
+        // Delete is async.
+        assertBusy(() -> {
+            try {
+                int actualFileCount = getFileCount(indexPath);
+                if (numberOfIterations <= RemoteStoreRefreshListener.LAST_N_METADATA_FILES_TO_KEEP) {
+                    assertEquals(numberOfIterations, actualFileCount);
+                } else {
+                    // As delete is async its possible that the file gets created before the deletion or after
+                    // deletion.
+                    assertTrue(actualFileCount >= 10 || actualFileCount <= 11);
```

Comment:
Why is there a possibility of the file count being 11? `assertBusy` should cover the async nature of deletion.

I think count being 11 is probably due to flush happening before the indexing. If we move this to after indexing, we can have deterministic check here -
https://github.com/opensearch-project/OpenSearch/blob/17ee1ce11d7b4382418454b9319679b9a5e4cbe9/server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java#L65-L84